pkg/oauth/api/validation/validation.go

```diff
@@ -294,3 +294,19 @@ func ValidateScopes(scopes []string, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+func ValidateSerializedRedirectReference(sref *api.SerializedRedirectReference) field.ErrorList {
```

Comment:
Wouldn't it be easier to just exclude the object from the validation test?  Also, @deads2k wanted to use the validation to give a useful error message when the user tries to annotate a SA - problem is that SA stuff is upstream and wouldn't know about a Route.